gobblin-data-management/src/main/java/gobblin/data/management/retention/action/AccessControlAction.java

```diff
@@ -0,0 +1,133 @@
+import gobblin.data.management.retention.dataset.ConfigurableCleanableDataset;
+import gobblin.data.management.version.DatasetVersion;
+import gobblin.data.management.version.FileSystemDatasetVersion;
+import gobblin.util.ConfigUtils;
+import gobblin.util.reflection.GobblinConstructorUtils;
+
+
+/**
+ * A {@link RetentionAction} that is used to change the permissions/owner/group of a {@link FileSystemDatasetVersion}
+ */
+@Slf4j
+public class AccessControlAction extends RetentionAction {
+
+  /**
+   * Optional - The permission mode to set on selected versions. E.g 750
+   */
+  private static final String MODE_KEY = "mode";
+  /**
+   * Optional - The owner to set on selected versions
+   */
+  private static final String OWNER_KEY = "owner";
+  /**
+   * Optional - The group to set on selected versions
+   */
+  private static final String GROUP_KEY = "group";
+
+  private final Optional<String> permission;
+  private final Optional<String> owner;
+  private final Optional<String> group;
+  private final VersionSelectionPolicy<DatasetVersion> selectionPolicy;
+
+  public AccessControlAction(Config actionConfig, FileSystem fs) {
+    super(actionConfig, fs);
+    this.permission = Optional.fromNullable(ConfigUtils.getString(actionConfig, MODE_KEY, null));
+    this.owner = Optional.fromNullable(ConfigUtils.getString(actionConfig, OWNER_KEY, null));
+    this.group = Optional.fromNullable(ConfigUtils.getString(actionConfig, GROUP_KEY, null));
+    this.selectionPolicy = createSelectionPolicy(actionConfig);
+  }
+
+  /**
+   * Applies {@link #selectionPolicy} on <code>allVersions</code> and modifies permission/owner to the selected {@link DatasetVersion}s
+   * where necessary.
+   * <p>
+   * This action only available for {@link FileSystemDatasetVersion}. It simply skips the operation if a different type
+   * of {@link DatasetVersion} is passed.
+   * </p>
+   * {@inheritDoc}
+   * @see gobblin.data.management.retention.action.RetentionAction#execute(java.util.List)
+   */
+  @Override
+  public void execute(List<DatasetVersion> allVersions) throws IOException {
+
+    // Select version on which access control actions need to performed
+    for (DatasetVersion datasetVersion : this.selectionPolicy.listSelectedVersions(allVersions)) {
+
+      // Perform action if it is a FileSystemDatasetVersion
+      if (datasetVersion instanceof FileSystemDatasetVersion) {
+        FileSystemDatasetVersion fsDatasetVersion = (FileSystemDatasetVersion) datasetVersion;
+        for (Path path : fsDatasetVersion.getPaths()) {
+
+          if (this.fs.exists(path)) {
+
+            // Update permissions if set in config
+            if (this.permission.isPresent()) {
+              FsPermission fsPermission = new FsPermission(this.permission.get());
+              this.fs.setPermission(path, fsPermission);
+              log.debug(String.format("Set permissions for %s to %s", path.toString(), fsPermission));
```

Comment:
log.debug("Set permissions for {} to {}", path, fsPermission).

Thus you don't need to pay the cost of generating the string if logging is disabled.